hpx/config/compiler_specific.hpp

```diff
@@ -60,11 +60,12 @@
 #endif
 
 #if defined(_MSC_VER)
-#   define HPX_MSVC _MSC_VER
-#   define HPX_WINDOWS
-#   if defined(__NVCC__)
-#       define HPX_SINGLE_INHERITANCE __single_inheritance
-#   endif
+#  define HPX_MSVC _MSC_VER
+#  define HPX_WINDOWS
+#  if defined(__NVCC__)
+#    define HPX_SINGLE_INHERITANCE __single_inheritance
+#  endif
+#  define HPX_CDECL __cdecl
```

Comment:
The compiler flags for MSVC added by VC include a global `__vectorcall` calling convention for all functions. We need to be able to force `cdecl` for certain functions, though. This macro is used for those.